Zend/zend_API.c

```diff
@@ -429,6 +429,9 @@ ZEND_API bool ZEND_FASTCALL zend_parse_arg_class(zval *arg, zend_class_entry **p
 ZEND_API bool ZEND_FASTCALL zend_parse_arg_bool_weak(zval *arg, zend_bool *dest) /* {{{ */
 {
 	if (EXPECTED(Z_TYPE_P(arg) <= IS_STRING)) {
+		if (UNEXPECTED(Z_TYPE_P(arg) == IS_NULL)) {
+			zend_error(E_DEPRECATED, "Passing null to argument of type bool is deprecated");
```

Comment:
The error message is not great, but I haven't found a way to make it better. We don't have access to the argument number here and thus can print neither the argument name nor the full type of the argument. The information could be passed down, but then these checks wouldn't be free anymore.

Reply:
What about
```suggestion
			zend_error(E_DEPRECATED, "Argument of type bool no longer accept 'null'");
```
? As the error is already a `Deprecated` one, maybe it's redundant to say it on the message. Another suggestion is to give the user a hint on that to do:

```suggestion
			zend_error(E_DEPRECATED, "Argument of type bool no longer accept 'null'. Make it a union type if needed");
```

Reply:
Talking about making it a union is non-sense as we are talking about internal functions on which the end user has no say.